File: popuniv-spring/src/main/java/com/kiworld/popuniv/dto/UserResponse.java
package com.kiworld.popuniv.dto;

import com.kiworld.popuniv.entity.Role;
import com.kiworld.popuniv.entity.User;
import com.kiworld.popuniv.entity.UserGroup;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
@Builder
public class UserResponse {
    private long id;
    private String email;
    private String nickname;
    private Role role;
    private int groupId;
}

File: popuniv-kotlin/src/main/kotlin/org/example/popunivkotlin/dto/UserResponse.kt
package org.example.popunivkotlin.dto

import org.example.popunivkotlin.entity.University
import org.example.popunivkotlin.entity.User

class LoginResponse (
    val email: String,
    val nickname: String,
    val token: String
)

class UserInfoResponse (
    // id, email, nickname, role, group
    val id: Long,
    val email: String,
    val nickname: String,
    val role: String,
    val university: University
) {
    companion object {
        fun from(user: User) = UserInfoResponse(
            id = user.id,
            email = user.email,
            nickname = user.nickname,
            role = user.role.name,
            university = user.university
        )
    }
}